pyro/distributions/lkj.py

```diff
@@ -1,120 +1,19 @@
-        if d <= 1:
-            raise ValueError("d must be > 1 in any correlation matrix")
-        eta = eta.squeeze()
-        vector_size = (d * (d - 1)) // 2
-        alpha = eta.add(0.5 * (d - 1.0))
-
-        concentrations = torch.empty(vector_size, dtype=eta.dtype, device=eta.device)
-        i = 0
-        for k in range(d - 1):
-            alpha -= .5
-            concentrations[..., i:(i + d - k - 1)] = alpha
-            i += d - k - 1
-        self._gen = Beta(concentrations, concentrations)
-        self.eta = eta
-        self._d = d
-        self._lkj_constant = None
-        super().__init__(torch.Size(), torch.Size((d, d)), validate_args=validate_args)
-
-    def sample(self, sample_shape=torch.Size()):
-        with torch.no_grad():
-            y = self._gen.sample(sample_shape=sample_shape + self.batch_shape)
-        z = y.mul(2).add(-1.0)
-        return _vector_to_l_cholesky(z)
-
-    def expand(self, batch_shape, _instance=None):
-        new = self._get_checked_instance(LKJCorrCholesky, _instance)
-        batch_shape = torch.Size(batch_shape)
-        new._gen = self._gen
-        new.eta = self.eta
-        new._d = self._d
-        new._lkj_constant = self._lkj_constant
-        super(LKJCorrCholesky, new).__init__(batch_shape, self.event_shape, validate_args=False)
-        new._validate_args = self._validate_args
-        return new
-
-    def lkj_constant(self, eta, K):
-        if self._lkj_constant is not None:
-            return self._lkj_constant
-
-        Km1 = K - 1
-
-        constant = torch.lgamma(eta.add(0.5 * Km1)).mul(Km1)
-
-        k = torch.linspace(start=1, end=Km1, steps=Km1, dtype=eta.dtype, device=eta.device)
-        constant -= (k.mul(math.log(math.pi) * 0.5) + torch.lgamma(eta.add(0.5 * (Km1 - k)))).sum()
-
-        self._lkj_constant = constant
-        return constant
-
-    def log_prob(self, x):
-        if self._validate_args:
-            self._validate_sample(x)
-
-        eta = self.eta
-
-        lp = self.lkj_constant(eta, self._d)
-
-        Km1 = self._d - 1
-
-        log_diagonals = x.diagonal(offset=0, dim1=-1, dim2=-2)[..., 1:].log()
-        # TODO: Figure out why the `device` kwarg to torch.linspace seems to not work in certain situations,
-        # and a seemingly redundant .to(x.device) is needed below.
-        values = log_diagonals * torch.linspace(start=Km1 - 1, end=0, steps=Km1,
-                                                dtype=x.dtype,
-                                                device=x.device).expand_as(log_diagonals).to(x.device)
-
-        values += log_diagonals.mul(eta.mul(2).add(-2.0))
-        values = values.sum(-1) + lp
-        values, _ = torch.broadcast_tensors(values, torch.empty(self.batch_shape))
-        return values
+        raise FutureWarning('class LKJCorrCholesky(d, eta, validate_args=None) is deprecated ' +
+                            'in favor of LKJCholesky(dim, concentration, validate_args=None).')
```

Comment:
The arg names for the pytorch distribution match LKJ and numpyro, so I have added a FutureWarning here. There are some other differences, e.g. concentration can be vectorized in the pytorch version but that's stricly less constrained than what we already have.

Reply:
I guess you want to use `warnings.warn(..., FutureWarning)` here?